src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -37,11 +38,51 @@
         "Error fetching cristin project with id: %s . Exception Message: %s";
     private static final String ERROR_MESSAGE_BACKEND_FETCH_FAILED =
         "Your request cannot be processed at this time due to an upstream error";
+    private static final String ERROR_MESSAGE_NOT_FOUND =
+        "The requested resource %s does not exist";
+    private static final String ERROR_MESSAGE_INTERNAL_ERROR =
+        "Your request cannot be processed at this time because of an internal server error";
+    private static final HttpClient client = HttpClient.newHttpClient();
+    private static final int STATUS_CODE_NOT_FOUND = 404;
+    private static final int STATUS_CODE_START_OF_ERROR_CODES_RANGE = 299;
+    private static final String CRISTIN_PROJECT_MATCHING_ID_IS_NOT_VALID =
+        "Project matching id %s does not have valid data";
 
     protected static <T> T fromJson(InputStreamReader reader, Class<T> classOfT) throws IOException {
         return OBJECT_MAPPER.readValue(reader, classOfT);
     }
 
+    protected static <T> T fromJson(InputStream stream, Class<T> classOfT) throws IOException {
+        return fromJson(new InputStreamReader(stream, StandardCharsets.UTF_8), classOfT);
+    }
+
+    /**
+     * Creates a NvaProject object containing a single transformed Cristin Project. Is used for serialization to the
+     * client.
+     *
+     * @param id       The Cristin id of the project to query
+     * @param language Language used for some properties in Cristin API response
+     * @return a NvaProject filled with one transformed Cristin Project
+     * @throws BadGatewayException          when there is a problem with fetch from backend
+     * @throws InternalServerErrorException when there is an unforeseen error
+     * @throws NotFoundException            when upstream returns 404
+     */
+    public NvaProject queryOneCristinProjectUsingIdIntoNvaProject(String id, String language)
+        throws BadGatewayException, InternalServerErrorException, NotFoundException {
+
+        CristinProject cristinProject = attemptToGetCristinProject(id, language);
+
+        if (cristinProject == null || !cristinProject.hasValidContent()) {
```

Comment:
The check-null-check-value pattern for me typically insinuates that the preceding call should return optional and the body of the if is an Optional::orElseThrow.